src/components/DropzoneArea.js

```diff
@@ -17,117 +10,38 @@ const splitDropzoneAreaProps = (props) => {
-
-        if (onChange) {
-            onChange(fileObjects.map((fileObject) => fileObject.file));
-        }
-    }
-
-    loadInitialFiles = async() => {
-        const {initialFiles} = this.props;
-        try {
-            const fileObjs = await Promise.all(
-                initialFiles.map(async(url) => {
-                    const file = await createFileFromUrl(url);
-                    const data = await readFile(file);
-
-                    return {
-                        file,
-                        data,
-                    };
-                })
-            );
 
-            this.setState((state) => ({
-                fileObjects: [].concat(
-                    state.fileObjects,
-                    fileObjs
-                ),
-            }), this.notifyFileChange);
-        } catch (err) {
-            console.log(err);
-        }
-    }
-
-    addFiles = async(newFileObjects) => {
-        const {filesLimit} = this.props;
-        // Update component state
-        this.setState((state) => {
-            // Handle a single file
-            if (filesLimit <= 1) {
-                return {
-                    fileObjects: [].concat(newFileObjects[0]),
-                };
-            }
-
-            // Handle multiple files
-            return {
-                fileObjects: [].concat(
-                    state.fileObjects,
-                    newFileObjects
-                ),
-            };
-        }, this.notifyFileChange);
-    }
-
-    deleteFile = (removedFileObj, removedFileObjIdx) => {
-        event.stopPropagation();
-
-        const {onDelete} = this.props;
-        const {fileObjects} = this.state;
-
-        // Calculate remaining fileObjects array
-        const remainingFileObjs = fileObjects.filter((fileObject, i) => {
-            return i !== removedFileObjIdx;
-        });
-
-        // Notify removed file
-        if (onDelete) {
-            onDelete(removedFileObj.file);
-        }
-
-        // Update local state
-        this.setState({
-            fileObjects: remainingFileObjs,
-        }, this.notifyFileChange);
-    }
-
-    render() {
-        const [, dropzoneAreaProps] = splitDropzoneAreaProps(this.props);
-        const {fileObjects} = this.state;
-
-        return (
-            <DropzoneAreaBase
-                {...dropzoneAreaProps}
-                fileObjects={fileObjects}
-                onAdd={this.addFiles}
-                onDelete={this.deleteFile}
-            />
-        );
-    }
-}
+const DropzoneArea = ({
+    clearOnUnmount,
+    initialFiles,
+    onChange,
+    onDelete,
+    filesLimit,
+    ...dropzoneAreaBaseProps
+
+}) => {
+    const {
+        handleDeleteFile,
+        handleAddFiles,
+        fileObjects,
+    } = useFiles({
+        onChange,
+        clearOnUnmount,
+        initialFiles,
+        filesLimit,
+        onDelete,
+    });
+
+    return (
+        <DropzoneAreaBase
+            {...dropzoneAreaBaseProps}
```

Comment:
I originally was explicit with this list, but whilst we're still porting stuff in, this leaves us open to a maintainance nightmare i.e. the onAlert stuff would have needed to be passed through etc... so for now using this pattern